src/servers/src/http/authorize.rs

```diff
@@ -12,118 +12,108 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::marker::PhantomData;
-
 use ::auth::UserProviderRef;
+use axum::extract::State;
 use axum::http::{self, Request, StatusCode};
-use axum::response::Response;
+use axum::middleware::Next;
+use axum::response::{IntoResponse, Response};
+use axum::Json;
 use base64::prelude::BASE64_STANDARD;
 use base64::Engine;
 use common_catalog::consts::DEFAULT_SCHEMA_NAME;
 use common_catalog::parse_catalog_and_schema_from_db_string;
 use common_error::ext::ErrorExt;
 use common_telemetry::warn;
-use futures::future::BoxFuture;
 use headers::Header;
-use http_body::Body;
 use secrecy::SecretString;
 use session::context::QueryContext;
 use snafu::{ensure, OptionExt, ResultExt};
-use tower_http::auth::AsyncAuthorizeRequest;
 
 use super::header::GreptimeDbName;
-use super::PUBLIC_APIS;
+use super::{JsonResponse, PUBLIC_APIS};
 use crate::error::{
     self, InvalidAuthorizationHeaderSnafu, InvalidParameterSnafu, InvisibleASCIISnafu,
     NotFoundInfluxAuthSnafu, Result, UnsupportedAuthSchemeSnafu, UrlDecodeSnafu,
 };
 use crate::http::HTTP_API_PREFIX;
 
-pub struct HttpAuth<RespBody> {
-    user_provider: Option<UserProviderRef>,
-    _ty: PhantomData<RespBody>,
+#[derive(Clone)]
+pub struct AuthState {
+    pub user_provider: Option<UserProviderRef>,
 }
 
-impl<RespBody> HttpAuth<RespBody> {
-    pub fn new(user_provider: Option<UserProviderRef>) -> Self {
-        Self {
-            user_provider,
-            _ty: PhantomData,
+pub async fn inner_auth<B>(
+    user_provider: Option<UserProviderRef>,
+    mut req: Request<B>,
+) -> std::result::Result<Request<B>, Response> {
+    // 1. prepare
+    let (catalog, schema) = extract_catalog_and_schema(&req);
```

Comment:
Here I suggest we return None when db parameter is not given so in error message we don't need to include default `greptime-public` when user doesn't provide `db`.

Reply:
You mean authentication stops immediately when user provider is presented but no `db` parameter is found, even in OSS cases?

Reply:
It's only about error message. If user did not provide `db` parameter, our error message should say `username or password mismatch` without mentioning the default value `greptime-public`.

Reply:
On second thought, I would like to move this part of refactor to a separate pr, where I can rework on authentication interface for it to meet both OSS and other cases. Resolving the conversation for now.